File: templates/matlab/seasonal_subseries_v2.m
function fig = seasonal_subseries_v2()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme(); rng(21);
    months = {'Jan','Feb','Mar','Apr','May','Jun', ...
              'Jul','Aug','Sep','Oct','Nov','Dec'};
    n_years = 6;
    season = 10 + 4 * sin(((0:11) - 2) * pi / 6);
    M = repmat(season, n_years, 1) + 0.5 * (0:n_years-1)' ...
        + 0.6 * randn(n_years, 12);                  % years x months
    yl = [min(M(:)) - 0.6, max(M(:)) + 0.6];
    fig = figure('Position',[100 100 900 320]);
    for m = 1:12
        subplot(1, 12, m); hold on;
        plot(0:n_years-1, M(:, m), '-o', 'Color', palette('cat',1), ...
             'MarkerFaceColor', palette('cat',1), 'MarkerSize', 2.5, ...
             'LineWidth', 1);
        mu = mean(M(:, m));
        plot([-0.4 n_years-0.6], [mu mu], 'Color', palette('cat',2), ...
             'LineWidth', 1.4);
        ylim(yl); xlim([-0.5 n_years-0.5]);
        set(gca, 'XTick', []);
        if m > 1, set(gca, 'YTickLabel', []); else, ylabel('value'); end
        xlabel(months{m}, 'FontSize', 7);
        set(gca, 'XGrid', 'off', 'YGrid', 'on');
    end
    sgtitle('Seasonal subseries (faceted by month)');
    annotation('textbox', [0 0 1 0.05], 'String', ...
               'year index within each month panel', 'FontSize', 8, ...
               'HorizontalAlignment', 'center', 'EdgeColor', 'none');
end

File: templates/python/seasonal_subseries_v2.py
"""seasonal_subseries_v2: 季节子序列分面图（12 个月分面小图 + 月均值线）."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme
from palette import cycle

def make_figure(title='Seasonal subseries (faceted by month)'):
    apply_theme(fig_size=(9, 3.2))
    rng = np.random.default_rng(21)
    months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
              'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
    n_years = 6
    season = 10 + 4 * np.sin((np.arange(12) - 2) * np.pi / 6)
    M = season + 0.5 * np.arange(n_years)[:, None] \
        + rng.normal(0, 0.6, (n_years, 12))            # 年 x 月
    fig, axes = plt.subplots(1, 12, sharey=True)
    for m, ax in enumerate(axes):
        ax.plot(np.arange(n_years), M[:, m], '-o', color=cycle(0),
                markersize=2.5, linewidth=1)
        ax.axhline(M[:, m].mean(), color=cycle(1), linewidth=1.4)
        ax.set_xticks([])
        ax.set_xlabel(months[m], fontsize=7)
        ax.grid(True, axis='y', linestyle=':', alpha=0.5)
        for sp in ('top', 'right'):
            ax.spines[sp].set_visible(False)
    axes[0].set_ylabel('value')
    fig.suptitle(title)
    fig.supxlabel('year index within each month panel', fontsize=8)
    fig.tight_layout(w_pad=0.2)
    return fig

if __name__ == '__main__':
    fig = make_figure(); fig.savefig(__file__.replace('.py', '.png'), dpi=150)
